Doc/reference/datamodel.rst

```diff
@@ -2215,23 +2215,80 @@ case the instance is itself a class.
 Emulating generic types
 -----------------------
 
-One can implement the generic class syntax as specified by :pep:`484`
-(for example ``List[int]``) by defining a special method:
+When using :term:`type annotations<annotation>`, it is often useful to
+*parameterize* a generic type using Python's square-brackets notation.
+For example, the annotation ``list[int]`` might be used to signify a
+:class:`list` in which all the elements are of type :class:`int`.
+
+.. seealso::
+
+   :pep:`484` - Type Hints
+      Introducing Python's framework for type annotations
+
+   :ref:`Generic Alias Types<types-genericalias>`
+      Documentation for objects representing parameterized generic classes
+
+   :class:`typing.Generic`
+      Inherit from :class:`typing.Generic` to implement generic classes that
+      can be parameterized at runtime and understood by static type-checkers.
+
+A class can generally only be parameterized if it defines the special
+classmethod ``__class_getitem__()``.
 
 .. classmethod:: object.__class_getitem__(cls, key)
 
    Return an object representing the specialization of a generic class
    by type arguments found in *key*.
 
-This method is looked up on the class object itself, and when defined in
-the class body, this method is implicitly a class method.  Note, this
-mechanism is primarily reserved for use with static type hints, other usage
-is discouraged.
+``__getitem__`` *versus* ``__class_getitem__``
 
-.. seealso::
+   Usually, the :ref:`subscription <subscriptions>` of an object in Python
+   using the square-brackets notation will call the :meth:`~object.__getitem__`
+   instance method defined on the object's class.
 
-   :pep:`560` - Core support for typing module and generic types
+   For example, if we have a list ``food`` as follows::
+
+      food = ['spam', 'eggs', 'bacon']
+
+   Calling ``food[0]`` will return the same value as calling::
+
+      type(food).__getitem__(food, 0)
+
```

Comment:
IMO we can drop the example here. ``__getitem__`` is already explained somewhere else.